### PR TITLE
Add clarity in explanation of various app components in DG and UG

### LINKED ISSUE #366: [PE-D][Tester D] In the UI Component section, the terms "AppointmentListPanel," "MainWindow," and "CommandBox" are used but not defined. Adding a glossary entry or initial explanation would improve clarity.
-------------
Labels: `severity.Low` `type.DocumentationBug`
original: yuvrajaryan/ped#4

### LINKED ISSUE #379: [PE-D][Tester D] Explain the time format
It would be helpful to explain the time format HHMM.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: yuvrajaryan/ped#9

### PR DESCRIPTION
**Pull request by yadobler**
fixes #366 
fixes #379

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -208,7 +209,7 @@ the constraints of each parameter when used in a command.
...
+| `APPOINTMENT_TIME` | Appointment time of the patient.                                                                           | - Must be in the format `HHMM-HHMM` (24-hour clock interval where `HH` is hours from `00` to `23` and `MM` is minutes from `00` to `59`). <br> - Time format should be (start time - end time) with start time earlier than end time. <br> - Start and end times should not equal. <br> - Should not be blank.                                                                                                                                                     | :white_check_mark: `0900-1100` <br> :white_check_mark: `2330-2359` <br> :x: `0900-0900` <br> :x: `1400-1300` <br> :x: `2200-0000` |
```
Comment (yong-zhuo) [inline docs/UserGuide.md:212]: Good explanation which is more precise for readers 👍🏻

File: docs/UserGuide.md [hunk 2]
```diff
@@ -144,13 +144,14 @@ This example shows how a typical command might look in MediBase3, with three mai
...
+> edit S1234567A n/John Doe i/S8234567B d/2000-01-01 g/M p/98765432 
 > e/johnd@example.com a/311, Clementi Ave 2, #02-25
 >```
 
-- **Command:** `add`
+- **Command:** `edit`
+- **Preamble:** `S1234567A`
 - **Prefix:** `n/` (name), `i/` (NRIC), `d/` (date of birth), `g/` (gender), `p/` (phone), `e/` (email), `a/` (address)
-- **Parameter:** `John Doe`, `S1234567A`, `2000-01-01`, `M`, `98765432`, `johndoe@example.com`, `311, Clementi Ave 2, #02-25`
+- **Parameter:** `John Doe`, `S8234567B`, `2000-01-01`, `M`, `98765432`, `johndoe@example.com`, `311, Clementi Ave 2, #02-25`
```
Comment (yong-zhuo) [inline docs/UserGuide.md:154]: Good idea to show an example that uses preamble

**Review by yong-zhuo — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -724,19 +734,45 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * **Mainstream OS**: Windows, Linux, Unix, MacOS
 
-* **Medical Condition**: A diagnosis or health issue assigned to a patient, such as "Diabetes Type 2" or "Hypertension." This helps track and manage a patient's health status.
+* **Allergy**: A specific substance or condition that a patient has a sensitivity or adverse reaction to, such as "Peanuts" or "Lactose."
```
Comment (PresleyChew) [inline docs/DeveloperGuide.md:737]: Amend fullstop to be outside of lactose

**Review by PresleyChew — COMMENTED** [review]
LGTM Overall. Amend typo and will be ready for merge.

**Pull request merged**